bin/node-template/node/src/command.rs

```diff
@@ -14,36 +14,39 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-use sp_consensus_aura::sr25519::{AuthorityPair as AuraPair};
-use sc_cli::VersionInfo;
-use crate::service;
 use crate::chain_spec;
 use crate::cli::Cli;
+use crate::service;
+use sc_cli::{substrate_cli, SubstrateCli};
+use sp_consensus_aura::sr25519::AuthorityPair as AuraPair;
 
-/// Parse and run command line arguments
-pub fn run(version: VersionInfo) -> sc_cli::Result<()> {
-	let opt = sc_cli::from_args::<Cli>(&version);
+#[substrate_cli(
+	impl_name = "Substrate Node",
+	support_url = "support.anonymous.an",
+	copyright_start_year = 2017
+)]
+impl SubstrateCli for Cli {
+	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
+		Ok(match id {
+			"dev" => Box::new(chain_spec::development_config()),
+			"" | "local" => Box::new(chain_spec::local_testnet_config()),
+			path => Box::new(chain_spec::ChainSpec::from_json_file(std::path::PathBuf::from(path))?),
+		})
+	}
+}
 
-	let mut config = sc_service::Configuration::from_version(&version);
+/// Parse and run command line arguments
+pub fn run() -> sc_cli::Result<()> {
+	let cli = Cli::from_args();
 
-	match opt.subcommand {
+	match &cli.subcommand {
 		Some(subcommand) => {
-			subcommand.init(&version)?;
-			subcommand.update_config(&mut config, chain_spec::load_spec, &version)?;
-			subcommand.run(
-				config,
-				|config: _| Ok(new_full_start!(config).0),
-			)
-		},
+			let runtime = cli.create_runtime(subcommand)?;
+			runtime.run_subcommand(subcommand, |config| Ok(new_full_start!(config).0))
+		}
 		None => {
-			opt.run.init(&version)?;
-			opt.run.update_config(&mut config, chain_spec::load_spec, &version)?;
-			opt.run.run(
-				config,
-				service::new_light,
-				service::new_full,
-				&version,
-			)
-		},
+			let runtime = cli.create_runtime(&cli.run)?;
```

Comment:
No I guess it's a side effect to the fact that RunCmd is not part of Subcommand. We can solve this by adding a command "run" that we will need to type and that will run the node if you want :innocent: 

Or we can stop using structopt and use clap directly so we will have more flexibility. It will also solve the problem of the default values on the definitions (see [here](https://github.com/paritytech/substrate/pull/5271#discussion_r398017673)).

Or we can use structopt internals or debate with them if they should make augment_clap public or not.